File: mobile/src/api/weeklyPlan.ts
import { apiRequest } from "./client";

export type WeeklyPlanResponse = {
  user_id: string;
  week: { start: string; end: string };
  inputs: {
    active_resolutions: number;
    active_tasks_total: number;
    active_tasks_completed: number;
    completion_rate: number;
  };
  micro_resolution: {
    title: string;
    why_this: string;
    suggested_week_1_tasks: {
      title: string;
      duration_min: number | null;
      suggested_time: string | null;
    }[];
  };
  request_id: string;
};

type LatestResult = {
  plan: WeeklyPlanResponse | null;
  requestId: string | null;
  notFound: boolean;
};

export async function getWeeklyPlanLatest(userId: string): Promise<LatestResult> {
  try {
    const { data, response } = await apiRequest<WeeklyPlanResponse>(`/weekly-plan/latest?user_id=${userId}`);
    return { plan: data, requestId: data.request_id || response.headers.get("X-Request-Id"), notFound: false };
  } catch (error) {
    if (error instanceof Error && /404|not found|no weekly plan snapshot/i.test(error.message)) {
      return { plan: null, requestId: null, notFound: true };
    }
    throw error;
  }
}

export async function runWeeklyPlan(userId: string): Promise<{ plan: WeeklyPlanResponse; requestId: string | null }> {
  const { data, response } = await apiRequest<WeeklyPlanResponse>("/weekly-plan/run", {
    method: "POST",
    body: { user_id: userId },
  });
  return { plan: data, requestId: data.request_id || response.headers.get("X-Request-Id") };
}

File: backend/app/api/schemas/weekly_plan.py
"""Schemas for weekly plan preview."""
from __future__ import annotations

from datetime import date
from typing import List, Optional, Literal
from uuid import UUID

from pydantic import BaseModel, Field

class WeeklyPlanRunRequest(BaseModel):
    user_id: UUID

class WeekWindowPayload(BaseModel):
    start: date
    end: date


class WeeklyPlanInputs(BaseModel):
    active_resolutions: int
    active_tasks_total: int
    active_tasks_completed: int
    completion_rate: float


class SuggestedTaskPayload(BaseModel):
    title: str
    duration_min: Optional[int] = None
    suggested_time: Optional[Literal["morning", "afternoon", "evening"]] = None


class MicroResolutionPayload(BaseModel):
    title: str
    why_this: str
    suggested_week_1_tasks: List[SuggestedTaskPayload]


class WeeklyPlanPreviewResponse(BaseModel):
    user_id: UUID
    week: WeekWindowPayload
    inputs: WeeklyPlanInputs
    micro_resolution: MicroResolutionPayload
    request_id: str
